packages/compass-import-export/src/utils/collection-stream.ts

```diff
@@ -223,7 +227,22 @@ export class WritableCollectionStream extends Writable {
 
     this.emit('progress', progressStats);
 
-    return callback();
+    return callback(this._makeStreamError());
+  }
+
+  _makeStreamError(): CollectionStreamError | undefined {
+    if (this.stopOnErrors && this._errors.length) {
+      const error = this._errors[0];
+      if (error instanceof Error) {
+        return error;
+      }
+      return {
+        name: 'CollectionStreamError',
+        message: 'Something went wrong while writing data to a collection',
+        original: error,
```

Comment:
Would it make sense to name this `cause` instead of `original` to be consistent with the standard JS error property?